sim/shaman/lightning_bolt.go

```diff
@@ -85,25 +88,19 @@ func (shaman *Shaman) newLightningBoltSpell(isLightningOverload bool) *core.Spel
 		baseDamage := dmgBonus + sim.Roll(719, 819) + spellCoeff*spell.SpellPower()
 		result := spell.CalcDamage(sim, target, baseDamage, spell.OutcomeMagicHitAndCrit)
 
-		if result.Landed() {
```

Comment:
If you're going to get rid of the landed check this way, you should just have it exit early. If it did not land then you don't need that check below. Generally speaking I try to write early exits wherever possible as it reduces nesting ifs.

Reply:
Looking at the code, an early out doesn't make sense since DealDamage() has to called no matter what.

Reply:
I meant if it didn't land you can exit the function (since no damage is applied on a miss)

Reply:
its ok either way